nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -384,6 +385,126 @@ public void instantiationInUnannotatedCode() {
+            "   return a;",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void genericFunctionReturnTypeMultipleReturnStatementsIfElseBlock() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static class A<T extends @Nullable Object> { }",
+            "  static A<String> method1(A<@Nullable String> a, int num) {",
+            "   if (num % 2 == 0) {",
+            "    // BUG: Diagnostic contains: Cannot assign from type",
+            "     return a;",
+            "    } else {",
+            "     return new A<String>();",
+            "    }",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void genericFunctionReturnTypeMultipleReturnStatementsInsideLoop() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static class A<T extends @Nullable Object> { }",
+            "  static A<String> method1(A<@Nullable String> a, int num) {",
+            "   for(int i = 0; i < num; i++) {",
+            "     if (i % 13 == 0) {",
+            "      // BUG: Diagnostic contains: Cannot assign from type",
+            "       return a;",
+            "     }",
+            "    } ",
+            "    return new A<String>();",
+            "  }",
+            "  static A<String> method2(A<@Nullable String> a, int num) {",
+            "   while(num != 0) {",
+            "     if (num % 13 == 0) {",
+            "      // BUG: Diagnostic contains: Cannot assign from type",
+            "       return a;",
+            "     }",
+            "     num --;",
+            "    } ",
+            "    return new A<String>();",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void genericsChecksForTernaryOperator() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "static class A<T extends @Nullable Object> { }",
+            "  static void method1(boolean t) {",
+            "      // BUG: Diagnostic contains: Cannot assign from type",
+            "   A<@Nullable String> t1 = t ? new A<String>() : new A<@Nullable String>();",
+            "  }",
+            "  static void method2(boolean t) {",
+            "      // BUG: Diagnostic contains: Cannot assign from type",
+            "   A<@Nullable String> t1 = t ? new A<String>() : new A<String>();",
+            "   // BUG: Diagnostic contains: Cannot assign from type",
+            "   A<String> t2 = t ? new A<@Nullable String>() : new A<@Nullable String>();",
+            "  }",
+            "}")
+        .doTest();
```

Comment:
Can we add a test where there is a method with return type `A<String>` and there is a statement `return t ? new A<@Nullable String>() : new A<@Nullable String>();`?